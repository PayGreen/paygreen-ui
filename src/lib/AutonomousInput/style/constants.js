import { math } from 'polished';

const fieldHeight = '34px';
const fieldRadius = math(fieldHeight + '/2');

export {
    fieldHeight,
    fieldRadius
};