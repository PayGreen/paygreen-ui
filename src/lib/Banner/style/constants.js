const color = {
    first: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.primary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientBase']
    },
    second: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.secondary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientShade']
    }
};

export {
    color
};