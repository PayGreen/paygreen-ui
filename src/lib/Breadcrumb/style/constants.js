import { transparentize } from 'polished';

const textColor = {
    default: {
        original: props => props.theme.wab.grey30,
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    link: {
        original: props => transparentize(0.3, props.theme.color[props.colorTheme].main),
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    hover: {
        original: props => props.theme.color[props.colorTheme].main,
        reverse: props => transparentize(0.05, props.theme.wab.white00),
    }
};

export {
    textColor
};