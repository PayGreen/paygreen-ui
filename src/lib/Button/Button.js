import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
    colorOptions,
    colorDefault,
    buttonSizeOptions,
    buttonSizeDefault,
    buttonTemplateOptions,
    buttonTemplateDefault,
    colorTypeOptions,
    colorTypeDefault
} from '../../shared/constants';
import {
    ButtonFill,
    ButtonLine,
    ButtonFillReverse,
    ButtonLineReverse,
    ButtonFillDisabled,
    ButtonLineDisabled
} from './style';

class Button extends PureComponent {
    render() {
        if (this.props.disabled) {
            if (this.props.template === buttonTemplateOptions.line) {
                return <ButtonLineDisabled {...this.props}>{this.props.children}</ButtonLineDisabled>;
            }
            
            return <ButtonFillDisabled {...this.props}>{this.props.children}</ButtonFillDisabled>;
        } else if (this.props.colorType === colorTypeOptions.reverse) {
            if (this.props.template === buttonTemplateOptions.line) {
                return <ButtonLineReverse {...this.props}>{this.props.children}</ButtonLineReverse>;
            }
            
            return <ButtonFillReverse {...this.props}>{this.props.children}</ButtonFillReverse>;
        } else {
            // Case of this.props.colorType === colorTypeDefault
            if (this.props.template === buttonTemplateOptions.line) {
                return <ButtonLine {...this.props}>{this.props.children}</ButtonLine>;
            }
            
            return <ButtonFill {...this.props}>{this.props.children}</ButtonFill>;
        }
    }
}

Button.propTypes = {
    template: PropTypes.oneOf(Object.values(buttonTemplateOptions)),
    colorType: PropTypes.oneOf(Object.values(colorTypeOptions)),
    color: PropTypes.oneOf(Object.values(colorOptions)),
    size: PropTypes.oneOf(Object.values(buttonSizeOptions)),
    disabled: PropTypes.bool,
};

Button.defaultProps = {
    template: buttonTemplateDefault,
    colorType: colorTypeDefault,
    color: colorDefault,
    size: buttonSizeDefault,
    disabled: false,
};

export default Button;