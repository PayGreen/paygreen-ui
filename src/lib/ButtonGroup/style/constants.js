import { math } from 'polished';

const bottomCoeff = 0.75;

const buttonSpace = props => props.resetMargin ? props.theme.space.sm : 0;

const calculateSpace = (space, coeff = 1) => {
    return props => math(props.theme.blockSpace[space] + '*' + coeff + '-' + buttonSpace(props));
};

export {
    bottomCoeff,
    buttonSpace,
    calculateSpace
};