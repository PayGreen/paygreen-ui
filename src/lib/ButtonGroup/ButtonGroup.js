import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
    alignOptions,
    spaceOptions,
    spaceDefault,
} from '../../shared/constants';
import { ButtonGroupBase } from './style';

class ButtonGroup extends PureComponent {
    render() {
        return <ButtonGroupBase
            {...this.props}
        >
            {this.props.children}
        </ButtonGroupBase>;
    }
}

ButtonGroup.propTypes = {
    align: PropTypes.oneOf(Object.values(alignOptions)),
    resetMargin: PropTypes.bool,
    marginLateral: PropTypes.oneOf(Object.values(spaceOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
};

ButtonGroup.defaultProps = {
    align: alignOptions.center,
    resetMargin: true,
    marginLateral: spaceDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
};

export default ButtonGroup;