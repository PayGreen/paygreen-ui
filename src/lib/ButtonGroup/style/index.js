import styled from 'styled-components';
import { justifyContentOptions } from '../../../shared/constants';
import { responsiveSpaces } from '../../../shared/spaces';

const ButtonGroupBase = styled.div`
    display: flex;
    justify-content: ${props => justifyContentOptions[props.buttonAlign]};
    flex-wrap: wrap;
    ${props => responsiveSpaces(
        'margin',
        props.resetMargin ? props.theme.space.sm : 0,
        0.75
    )};

    & > * {
        margin: ${props => props.theme.space.sm};
    }
`;

export { ButtonGroupBase };