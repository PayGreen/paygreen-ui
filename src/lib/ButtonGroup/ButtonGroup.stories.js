import React from 'react';
import ButtonGroup from './ButtonGroup';
import Button from '../Button/Button';
import {
    buttonSizeOptions,
    buttonSizeDefault,
    buttonTemplateOptions,
    alignOptions,
    blockSpaceOptions,
    blockSpaceDefault
} from '../../shared/constants';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, radios, select } from '@storybook/addon-knobs';

storiesOf('ButtonGroup', module)
    .addDecorator(withKnobs)
    .add('ButtonGroup', () => (
        <ButtonGroup
            align={radios('Buttons align', alignOptions, alignOptions.center)}
            resetMargin={boolean('Compensate buttons margin', false)}
            marginLateral={select('Lateral margin', blockSpaceOptions, blockSpaceDefault)}
            marginTop={select('Margin top', blockSpaceOptions, blockSpaceDefault)}
            marginBottom={select('Margin bottom', blockSpaceOptions, blockSpaceDefault)}
        >
            <a href="#">
                <Button
                    template={buttonTemplateOptions.line}
                    buttonSize={radios('Buttons size', buttonSizeOptions, buttonSizeDefault)}
                >
                    First button
                </Button>
            </a>

            <a href="#">
                <Button
                    buttonSize={radios('Buttons size', buttonSizeOptions, buttonSizeDefault)}
                >
                    Second button
                </Button>
            </a>
        </ButtonGroup>
    ), {
        notes: 'You must always use Button component inside a ButtonGroup. You can add margin-top, margin-bottom, and padding (left and right only) with props. Button component comes with its own margins: in order to avoid shift with other components, you can compensate buttons margin with the resetMargin prop on ButtonGroup.',
    });