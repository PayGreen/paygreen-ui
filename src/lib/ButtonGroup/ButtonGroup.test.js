import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import { buttonStyleOptions } from '../../shared/constants';
import Button from '../Button/Button';
import ButtonGroup from './ButtonGroup';

it('renders without crashing', () => {
    const buttonGroup = TestRenderer.create(
        <ButtonGroup theme={ThemeDefault}>
            <a href="#">
                <Button
                    theme={ThemeDefault}
                    template={buttonStyleOptions.line}
                >
                    First button
                </Button>
            </a>

            <a href="#">
                <Button theme={ThemeDefault}>
                    Second button
                </Button>
            </a>
        </ButtonGroup>
    );
    expect(buttonGroup.toJSON()).toMatchSnapshot();
});