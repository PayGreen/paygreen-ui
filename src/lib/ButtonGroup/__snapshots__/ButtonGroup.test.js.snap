// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__ButtonGroupBase-cn1823-0 cWBQvF"
>
  <button
    type="button"
  >
    <span
      className="style__ButtonBase-pnyves-0 godPhX"
    >
      First button
    </span>
  </button>
  <button
    type="button"
  >
    <span
      className="style__ButtonBase-pnyves-0 cuIwsX"
    >
      Second button
    </span>
  </button>
</div>
`;
